horovod/mxnet/mpi_ops.cc

```diff
@@ -202,68 +202,58 @@ extern "C" int horovod_mxnet_allreduce_async(
       RunContext rctx, Engine::CallbackOnComplete cb) mutable {
       DoAllreduce(input, output, average, new_name, cb);
   };
+#if HAVE_CUDA
   auto allreduce_async_cpu_fn = [input, output, new_name, average](
       RunContext rctx, Engine::CallbackOnComplete cb) mutable {
       DoAllreduceCudaOnCPU(input, output, average, new_name, cb);
   };
-  int cpu = -1;
-#ifdef HOROVOD_GPU_ALLREDUCE
-  if (input->ctx().dev_mask() == gpu::kDevMask &&
```

Comment:
Does the removal of this if-statement cause any problems? Was it a bug to include the if-statement here in the first place?

Reply:
This piece of code puzzles me as well. It seems the purpose is not for functionality by performance concern? Did you add this check here? I don't find similar checking in PyTorch and Tensorflow, thus wondering if this is indeed needed.

Reply:
Yes, I added it. I just remembered this is a layover of when we were still using CUDA-aware MPI. My reasoning back then was I wanted it to skip the copy-to-CPU when both input and output were on GPU. Since we are not using CUDA-aware MPI anymore, we can get rid of it.